lib/file-handling/file-demuxer.js

```diff
@@ -153,36 +153,41 @@ FileDemuxer.prototype._needsNewOutputStream = function() {
  * @returns {Number} shardSize
  */
 FileDemuxer.getOptimalShardSize = function(fileInfo, acc) {
-  var accumulator = typeof acc === 'undefined' ? 0 : acc;
-
-  // Determine hops back by accumulator
-  var hops = (accumulator - FileDemuxer.SHARD_MULTIPLES_BACK) < 0 ?
-                 0 :
-                 accumulator - FileDemuxer.SHARD_MULTIPLES_BACK;
-
-  // Calculate bytemultiple shard size by hops back
-  var shardSize = function(hops) {
-    return (8 * (1024 * 1024)) * Math.pow(2, hops);
-  };
-
-  var byteMultiple = shardSize(accumulator);
-  var check = fileInfo.fileSize / byteMultiple;
-
-  // Determine if bytemultiple is highest bytemultiple that is still <= fileSize
-  if (check > 0 && check <= 1) {
-
-    // Certify the number of concurrency * shardSize doesn't exceed freemem
-    while (
-      hops > 0 &&
-      (os.freemem() / shardSize(hops) <= fileInfo.shardConcurrency)
-    ) {
-      hops = hops - 1 <= 0 ? 0 : hops - 1;
+  if (typeof window === 'undefined') {
+    var shardSize = (8 * (1024*1024))
+    return shardSize;
+  } else {
+    var accumulator = typeof acc === 'undefined' ? 0 : acc;
+
+    // Determine hops back by accumulator
+    var hops = (accumulator - FileDemuxer.SHARD_MULTIPLES_BACK) < 0 ?
+                   0 :
+                   accumulator - FileDemuxer.SHARD_MULTIPLES_BACK;
+
+    // Calculate bytemultiple shard size by hops back
+    var shardSize = function(hops) {
+      return (8 * (1024 * 1024)) * Math.pow(2, hops);
+    };
+
+    var byteMultiple = shardSize(accumulator);
+    var check = fileInfo.fileSize / byteMultiple;
+
+    // Determine if bytemultiple is highest bytemultiple that is still <= fileSize
+    if (check > 0 && check <= 1) {
+
+      // Certify the number of concurrency * shardSize doesn't exceed freemem
+      while (
+        hops > 0 &&
+        (os.freemem() / shardSize(hops) <= fileInfo.shardConcurrency)
```

Comment:
I dislike the idea of having special browser logic in `core` but I'm unsure we will be able to get around it here.

Perhaps instead of `window === undefined` we could do something like `if(os.freemem === undefined)`